docs/SOURCING-COMMITMENTS.md

```diff
@@ -0,0 +1,91 @@
+
+## Source of commitment utxos
+
+Usually they will be sourced from your Joinmarket wallet, and this will require no intervention. The main purpose of this page is (a) to make that happen as much as possible and (b) to tell you what to do if it goes wrong.
+
+### Wait for at least 5 confirmations
+
+Commitments will not be accepted if the age is less than the default 5 confirmations, so **for a fresh wallet, wait for at least 5 blocks before starting as a user/taker**.
+
+### Fund your wallet with multiple utxos.
+
+To help, **when you fund your Joinmarket wallet** (as a Taker, i.e. a transaction originator using `sendpayment` or `tumbler`; for Makers it doesn't matter), it's recommended to **send the coins to multiple (e.g. 3) addresses**. In wallets like Bitcoin Core or Electrum (or I presume many others) this can be done in the GUI using the `sendtomany` or `paytomany` feature; you just enter multiple addresses (and probably pay the same amount to each one). If you're using Joinmarket sendpayment, you'll want to choose several receiving addresses in one external branch for one mixdepth. **Don't do a `sendtomany` to *different* mixdepths**, as that will link those mixdepths together, a central thing that Joinmarket avoids. Also, **don't send multiple outputs to the same address** - use different addresses.
+
+As an example, if you send to 3 address instead of 1, you will then have 9 commiments available - effectively 9 tries at doing the first transaction, instead of only 3 tries if you send the coins to one address.
+
+However, **don't split an amount to be spent into > 5** - because a utxo has to be at least 20% of the total payment amount, so you could find yourself unable to source a commitment in that case. 3 equal sized amounts ought to be fine.
+
+In most cases this will be enough.
+
+### Waiting longer for the tumbler.
+
+Because of the way the tumbler algorithm works, your utxos will get "used up" as coins are swept from one depth to the next, so it's nearly inevitable that at some point, the utxos you have are too new (<5 blocks) to provide a valid commitment. For this reason, the code now waits until you have a valid commitment in the wallet (see the switch `jm_single().wait_for_commitments = 1` in the tumbler.py code). This will mean that you can't run the tumbler relatively fast, e.g. if you did a run that *should* take around 4 hours, it might take several hours longer. If you use default or slower waits, it might not make any observable difference.
```

Comment:
There isn't `jm_single().wait_for_commitments` in tumbler.py code anymore (that wiki was written before clientserver rewrite of JM I think). Actually, searching through codebase I see `global_singleton.wait_for_commitments = 0` and that's all, it's never accessed in any way.